### ISSUE TITLE
`<ranges>`: `ranges::to` use range-based for loops in append branches

### ISSUE THREAD

**Issue by hewillk** [llvm/llvm-project#119133]
https://github.com/llvm/llvm-project/blob/6c062afc2e6ed4329e1e14cb011913195a5356fa/libcxx/include/__ranges/to.h#L112-L124

Range-based for loop is not guaranteed to be well-formed for `input_range` as it does not extract iterator-pair through `ranges::begin`/`ranges::end`:

https://godbolt.org/z/Kc3dsvnfW

```cpp
#include <ranges>

struct Vector {
  void push_back(int);
};

struct OnlyADLRange {
  void begin() = delete;
  void end() = delete;
  friend int* begin(OnlyADLRange&);
  friend int* end(OnlyADLRange&);
};

int main() {
  std::ranges::contiguous_range auto r = OnlyADLRange{};
  auto v = r | std::ranges::to<Vector>(); // only well-formed in libstdc++
}
```

**Comment by hewillk** [comment]
`range_formatter` have similar issues:

https://github.com/llvm/llvm-project/blob/4228a6ef2b41fe4c0c9f65839e2b7f4a0b4c6627/libcxx/include/__format/range_formatter.h#L197-L201

**Comment by hewillk** [comment]
It looks like libc++ extensively uses range-for loops to iterate over ranges when implementing P1206, so this should be something that can be modified uniformly (if worth it).

**Comment by frederick-vs-ja** [comment]
I think I can fix these issues (if wanted). But they mean that we can't traverse a `range` using range-`for` in generic code, which sounds weird.

**Comment by philnik777** [comment]
Yeah, I think there should be an LWG or CWG issue about this. Not being able to iterate a range with a range-based for loop sounds a lot like a design flaw to me.

**Comment by frederick-vs-ja** [comment]
[LWG4389](https://cplusplus.github.io/LWG/issue4389) submitted.
